contents/performance/Performance.md

```diff
@@ -0,0 +1,102 @@
+# Introduction to Performance Optimization
+
+Author: [Ian Teo](https://github.com/IanTeo)
+
+## Overview
+
+Computer performance can be defined as the rate of work accomplished by a computer system. Even if execution time is not important for a particular application, it may be important to reduce CPU cycles so as to consume less power; from applications running in small battery operated devices to huge data centres.
+
+`Premature optimization is the root of all evil.`<br>
+I think many people have heard of this quote from Donald Knuth before. This quote is actually misinterpreted frequently, because of the lack of context. Here is the full quote:<br>
+`We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. Yet we should not pass up our opportunities in that critical 3%.`<br>
+You can find a nice explaination of the quote [here](http://wiki.c2.com/?PrematureOptimization).
+
+This is not a guide on how to optimize that 3%, but rather, to teach you standard optimization techniques that you should apply in all of your code, so that you do not create sub-optimal codes (premature pessimization). If you want to find out more about optimizing that 3%, you can find more at [Other Resources](#other-resources) below.
```

Comment:
Use `describe` or `explain` instead of `teach`? The latter sets up unrealistic expectations.